sql/mysqld.cc

```diff
@@ -5344,6 +5344,26 @@ static void test_lc_time_sz()
 #endif//DBUG_OFF
 
 
+static void signal_accept_connections()
```

Comment:
eh, it's not just "signal accept connections", `handle_connection_sockets()` is the main program loop, that's where the server spends all its time. Everything before it is startup, everything after it is shutdown.

If you want that code in a separate function — it needs a better name. And also, please, explain your reasoning for extracting it into a function, I don't quite understand it yet

Reply:
😊 I wasn't sure of a good name (I used the word "signal" here not in the Linux `signal` sense, but that wasn't a good choice clearly). Maybe just `enter_main_loop` or something like that would be clearer.  Yes, I understand that `handle_connections_sockets` is where we spend all of our time, that's where the `select` call is and the main program loop.  I moved this to a separate function to encapsulate the symmetry of setting `select_thread_in_use` to true (before entering the main loop) needs to be paired with setting it to false when exiting the main loop.  I didn't dig through the git history but I suspect that this value was set closer to the call to `handle_connections_sockets` in the past, but code was added that moved it away gradually.  This function is meant to show that `handle_connections_sockets ` and `select_thread_in_use` go together (mostly, but we have another place where it is cleared... anyway, step by step).

Reply:
may be `run_main_loop` or `execute_main_loop` or something along these lines? `perform_` or whatever?

Reply:
Good idea, I will go with `run_main_loop`.